requirements.txt

```diff
@@ -5,3 +5,5 @@ numpy-quaternion
 pillow
 scipy>=1.3.0
 tqdm
+torch
```

Comment:
I think we've avoided adding torch as a dependency so far. @erikwijmans -- thoughts?

Reply:
Yeah, we should continue to avoid it as a hard dependency.  @mpiseno we can have the dataset extractor class be API-compatible and then mix-in a torch dataset in the training examples or something.   It could also be a soft dependency (i.e. you will only need it if you want to use this code).

Reply:
The only think I actually use pytorch for is in the tests in tests/test_dataset_extraction. I could remove this without any changes to the ImageExtractor class, but I think @mathfac wanted to run the data through a pytorch model in testing?

Reply:
That's fine, we have a soft dependency on pytorch for some tests already (--with-cuda relies on torch in places), but that soft dependency doesn't need to be in requirements.txt

Reply:
Is it sufficient to simply remove torch from requirements.txt? Will the soft dependency during testing be handled appropriately even if i import torch in my tests/test_dataset_extraction.py file?

Reply:
Yeah, just removing it from requirements.txt is sufficient.  All of a "soft" dependency is in these terms is something you don't list in requirements.txt and the user will only need to install if they actually want to run certain code (as opposed to just importing the function).